### PR TITLE
test: device attribute provider and customerio test fixes

### PR DESCRIPTION
**Pull request by ami-aman**
helps: https://github.com/customerio/issues/issues/11984
On Linear: https://linear.app/customerio/issue/MOB-14/journeys-cdp-test-coverage

Includes:
- fixed test cases for device attribute provider with datapipeline module change
- fixed CustomerIO class test cases
- minor fixes

Complete each step to get your pull request merged in. [Learn more about the workflow this project uses](https://github.com/customerio/customerio-ios/blob/develop/docs/dev-notes/GIT-WORKFLOW.md). 
- [ ] [Assign members of your team](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to review the pull request. 
- [ ] Wait for pull request status checks to complete. If there are problems, fix them until you see that [all status checks are passing](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fsymfony.com%2Fdoc%2F4.3%2F_images%2Fdocs-pull-request-symfonycloud.png&f=1&nofb=1). 
- [ ] Wait until the pull request has been reviewed *and approved* by a teammate
- [ ] After code reviews are approved and you determine this PR is ready to merge, select *Squash and Merge* button on this screen, leave the title and description to the default values, then merge the PR.

### CODE REVIEW

File: Sources/Tracking/migration/DataPipelineMigrationAssistant.swift [hunk 1]
```diff
@@ -54,7 +54,7 @@ class DataPipelineMigrationAssistant {
         // Remove the task from the queue if the task has been processed successfully
         defer {
             if isProcessed {
-//                backgroundQueue.deleteProcessedTask(task)
+                backgroundQueue.deleteProcessedTask(task)
```
Comment (levibostian) [inline Sources/Tracking/migration/DataPipelineMigrationAssistant.swift:57]: Why was this originally commented out and why is it modified in this PR? 

I was expecting that this PR fixes tests. I currently do not see this change doesn't fit that description.

File: Tests/Tracking/CustomerIOTest.swift [hunk 2]
```diff
@@ -15,19 +13,20 @@ class CustomerIOTest: UnitTest {
...
+        XCTAssertFalse(implmentationMock.registerDeviceTokenCalled)
+    }
 
-        XCTAssertEqual(cleanupRepositoryMock.cleanupCallsCount, 1)
+    func test_initialize_givenPushDeviceTokenSet_expectRegisterDeviceTokenCalled() {
+        let pushDeviceToken = String.random
+        globalDataStoreMock.pushDeviceToken = pushDeviceToken
+        customerIO.postInitialize(diGraph: diGraph)
+        XCTAssertTrue(implmentationMock.registerDeviceTokenCalled)
     }
```
Comment (levibostian) [inline Tests/Tracking/CustomerIOTest.swift:31]: Looking at [the customerio class](https://github.com/customerio/customerio-ios/blob/shahroz/track-cleanup/Sources/Tracking/CustomerIO.swift), I don't see anything in that code about device tokens. 

Am I looking at the wrong class? I don't understand why there are device token tests written in this PR.

File: Sources/Tracking/migration/DataPipelineMigrationAssistant.swift (hunk 1 above)
Comment (ami-aman) [reply Sources/Tracking/migration/DataPipelineMigrationAssistant.swift:57]: Right. This PR focuses on test fixes but description has that this PR also includes minor fixes such as these.

File: Tests/Tracking/CustomerIOTest.swift (hunk 2 above)
Comment (ami-aman) [reply Tests/Tracking/CustomerIOTest.swift:31]: Please [look into this class](https://github.com/customerio/customerio-ios/blob/shahroz/track-cleanup/Sources/Common/CustomerIOInstance.swift) for the correct method call.

Comment (levibostian) [reply Tests/Tracking/CustomerIOTest.swift:31]: Oh, ok. 

The test file that was modified is `Tests/Tracking/CustomerIOTest.swift`. The [class you linked to](https://github.com/customerio/customerio-ios/blob/shahroz/track-cleanup/Sources/Common/CustomerIOInstance.swift?rgh-link-date=2024-01-09T17%3A50%3A29Z) is `Sources/Common/CustomerIOInstance.swift`. 

I was expecting that the file in the `Tests/` directory map to the same file in the `Sources/` directory. 

I can understand why this tests file was modified to remove the module hooks logic. However, I would expect that the newly added test functions be moved into a different file.

**Review by levibostian — CHANGES_REQUESTED** [review]
Rejecting for 1 [thread I consider blocking](https://github.com/customerio/customerio-ios/pull/471#discussion_r1447485369).

File: Tests/Tracking/CustomerIOTest.swift (hunk 2 above)
Comment (ami-aman) [reply Tests/Tracking/CustomerIOTest.swift:31]: After carefully looking at the difference between what you suggest and what has been done currently, I think it does make a lot value to bring this class to `Tests/Common`. The other method in this class also calls the same function and [links to this class](https://github.com/customerio/customerio-ios/blob/shahroz/track-cleanup/Sources/Common/CustomerIOInstance.swift?rgh-link-date=2024-01-09T17%3A50%3A29Z).

Thanks for pointing out.

Comment (ami-aman) [reply Tests/Tracking/CustomerIOTest.swift]: Find the change in the [latest commit](https://github.com/customerio/customerio-ios/pull/471/commits/40e31c4c213116ef3dc915e99d8a4b9f410359ef).

**Review by levibostian — APPROVED** [review]

**Review by Shahroz16 — APPROVED** [review]

**Pull request merged**